editor/server_api.hpp

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include "geometry/rect2d.hpp"
+
+#include "std/map.hpp"
+#include "std/string.hpp"
+
+namespace osm
+{
+
+/// All methods here are synchronous and need wrappers for async usage.
+/// TODO(AlexZ): Rewrite ServerAPI interface to accept XMLFeature.
+class ServerApi06
+{
+  string m_user;
+  string m_password;
+  string m_baseOsmServerUrl;
+
+public:
+  // k= and v= tags used in OSM.
+  using TKeyValueTags = map<string, string>;
+
+  /// Some nodes can't be deleted if they are used in ways or relations.
+  enum class DeleteResult
+  {
+    ESuccessfullyDeleted,
+    EFailed,
+    ECanNotBeDeleted
+  };
+
+  ServerApi06(string const & user, string const & password, string const & baseUrl = "http://api.openstreetmap.org");
+  /// @returns true if connection with OSM server was established, and user+password are valid.
+  bool CheckUserAndPassword() const;
+  /// @returns http server code for given url or negative value in case of error.
+  /// This function can be used to check if user did not confirm email validation link after registration.
+  /// For example, for http://www.openstreetmap.org/user/UserName 200 is returned if UserName was registered.
+  static int HttpCodeForUrl(string const & url);
+
+  /// Please use at least created_by=* and comment=* tags.
+  bool CreateChangeSet(TKeyValueTags && kvTags, uint64_t & outChangeSetId) const;
```

Comment:
In-place map

> On 18.12.2015, at 22.59, mgsergio <notifications@github.com> wrote:
> 
> In editor/server_api.hpp:
> 
> > +  {
> > +    ESuccessfullyDeleted,
> > +    EFailed,
> > +    ECanNotBeDeleted
> > +  };
> > +
> > +  ServerApi06(string const & user, string const & password, string const & baseUrl = "http://api.openstreetmap.org");
> > +  /// @returns true if connection with OSM server was established, and user+password are valid.
> > +  bool CheckUserAndPassword() const;
> > +  /// @returns http server code for given url or negative value in case of error.
> > +  /// This function can be used to check if user did not confirm email validation link after registration.
> > +  /// For example, for http://www.openstreetmap.org/user/UserName 200 is returned if UserName was registered.
> > +  static int HttpCodeForUrl(string const & url);
> > +
> > +  /// Please use at least created_by=* and comment=* tags.
> > +  bool CreateChangeSet(TKeyValueTags && kvTags, uint64_t & outChangeSetId) const;
> Тут ожидается правая ссылка?
> 
> —
> Reply to this email directly or view it on GitHub.
>